Assets/Scripts/SS3D/Systems/Inventory/EntityInventory.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using SS3D.Core.Systems.Inventory.Subsystem.Clothing;
+using UnityEngine;
+
+namespace SS3D.Core.Systems.Inventory
+{
+    /// <summary>
+    /// Base class for all entities's inventory
+    /// </summary>
+    public class EntityInventory : MonoBehaviour
+    {
+        [SerializeField] private List<InventorySlot> _slots;
+
+        /// <summary>
+        /// Gets clothing slots from inventory, use T as the specific type of slot
+        /// TODO: Proper cache of this information, considering slots can change at runtime, like adding an arm/other type of hand 
+        /// </summary>
+        /// <returns></returns>
+        public List<T> GetSlots<T>()
+        {
+            List<T> slots = new();
+
+            foreach (InventorySlot inventorySlot in _slots)
+            {
+                if (inventorySlot is T type)
+                {
+                    slots.Add(type);
+                }
+            }
+
+            return slots;
+        }
```

Comment:
You can avoid the need to allocate a list:
```suggestion
        public IEnumerable<T> GetSlots<T>() => inventorySlot.OfType<T>();
```
(not tested lol)